CMakeLists.txt

```diff
@@ -1,19 +1,17 @@
-cmake_minimum_required(VERSION 3.10)
-
+cmake_minimum_required(VERSION 3.14)
 project(stc VERSION 0.1.0)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
 set(Boost_USE_STATIC_LIBS ON)
 find_package(Boost 1.70 REQUIRED COMPONENTS program_options)
+include_directories(${Boost_INCLUDE_DIRS}) 
 
-add_executable(stc src/main.cpp
```

Comment:
What is the difference between add_executable here and add_library down below?

Reply:
Add library creates a named library that can be linked against elsewhere, but doesn't generate a runnable binary. add_executable generates something you can actually run.